### ISSUE TITLE
"Should be single zip" error uploading SonarScanner for MSBuild ZIP

### ISSUE THREAD

**Issue by alexangas** [JetBrains/TeamCity.SonarQubePlugin#54]
I'm attempting to install a newer version of the SonarScanner for MSBuild, downloaded [from here](https://github.com/SonarSource/sonar-scanner-msbuild/releases). Example - https://github.com/SonarSource/sonar-scanner-msbuild/releases/download/4.3.1.1372/sonar-scanner-msbuild-4.3.1.1372-net46.zip

It doesn't matter how I rename the file, I always get the error "Should be single zip, name should contain version, eg: sonar-scanner-msbuild.4.0.2.892.zip"

![image](https://user-images.githubusercontent.com/132198/44840617-53922480-ac39-11e8-8eb9-73ee8a0003ac.png)

Even if I rename the file to the exact same filename in the error message, or increment its last digit by 1, the same error comes back.

Running TeamCity 2018.1.1 (build 58406) on Windows.

Any ideas? (Also, thank you for updating this plugin for the MSBuild scanner!)

**Comment by Linfar** [comment]
Hello

Do you use the latest plugin version (#2017.1-18)?

**Comment by alexangas** [comment]
I did use that version. Let me try again.

**Comment by alexangas** [comment]
Unfortunately no luck. Is there any extra logging I can enable?

**Comment by Linfar** [comment]
I'll check it on Windows to check if this is platform specific problem. Meanwhile you can enable debug-all log profile, may be there will be some additional messages.

**Comment by Linfar** [comment]
Sorry for delay. I've found the issue, indeed it was platform specific. Please try [updated version](https://teamcity.jetbrains.com/repository/download/TeamCityPluginsByJetBrains_TeamCitySonarQubePlugin_Build20171x/1593319:id/sonar-plugin.zip)

**Comment by alexangas** [comment]
Thanks for being able to repro. The error has now changed (with file sonar-scanner-msbuild.4.3.1.1372.zip) to:

java.lang.IllegalArgumentException: Illegal character in opaque part at index 11: jar:file:C:\TeamCity\data\plugins\.tools\sonar-scanner-msbuild.4.3.1.1372.zip

**Comment by alexangas** [comment]
Fixed! Thank you 😄
